src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -1,13 +1,94 @@
 import React from 'react';
-import {getFollowing} from '../../../service/profile';
+import {Context} from '../../../context';
+import DiscoveryAction from '../../../context/actions/discoveryAction';
+import following from '../../../context/actions/following';
 import {getFollowedDomain} from '../../../service/domain';
+import {getFollowing} from '../../../service/profile';
 import {getFollowingTopic} from '../../../service/topics';
-import following from '../../../context/actions/following';
-import {Context} from '../../../context';
 
 const useDiscovery = () => {
   const [, followingDispatch] = React.useContext(Context).following;
   const [refreshing, setRefreshing] = React.useState(false);
+  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
+
+  const users = React.useMemo(() => {
+    return discovery.initialUsers.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery]);
+
+  const exhangeFollower = (newUserLists, willFollow, userId) => {
+    const indexUser = newUserLists.findIndex((item) =>
+      item.user ? item.user.user_id === userId : item.user_id === userId
+    );
+    if (newUserLists[indexUser].user) {
+      newUserLists[indexUser].user.following = !!willFollow;
+      newUserLists[indexUser].user.is_followed = !!willFollow;
+    } else {
+      newUserLists[indexUser].following = !!willFollow;
+      newUserLists[indexUser].is_followed = !!willFollow;
+    }
+    return newUserLists[indexUser];
+  };
+
+  const topicExchangeFollower = (newTopicLists, willFollow, topicId, fromTopicPage) => {
+    let indexTopic;
+    if (fromTopicPage) {
+      indexTopic = newTopicLists.findIndex((item) => item.name === topicId);
+    } else {
+      indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
+    }
+    newTopicLists[indexTopic].following = !!willFollow;
+    return newTopicLists[indexTopic];
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `topicExchangeFollower` function also directly modifies the `newTopicLists` array. Similar to the previous comment, you should avoid mutating state directly. 

```diff
-    let indexTopic;
+    let updatedTopicLists = [...newTopicLists];
+    let indexTopic;
    if (fromTopicPage) {
      indexTopic = updatedTopicLists.findIndex((item) => item.name === topicId);
    } else {
      indexTopic = updatedTopicLists.findIndex((item) => item.topic_id === topicId);
    }
-    newTopicLists[indexTopic].following = !!willFollow;
+    updatedTopicLists[indexTopic].following = !!willFollow;
-    return newTopicLists[indexTopic];
+    return updatedTopicLists[indexTopic];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->